vision/fiducial_detection_and_pose.hh

```diff
@@ -28,7 +28,7 @@ struct MarkerInWorld {
   int id;
 };
 
-std::vector<MarkerDetection> detect_markers(const cv::Mat& mat);
+std::vector<MarkerDetection> detect_markers(cv::Mat mat);
```

Comment:
One should strongly prefer to take things by const-ref, and then not mutate the object. It's a faux-pas in C++ to edit input arguments (pure output arguments are ok, but peligroso).

Consider: It's very hard for the caller to tell that the image is being edited, so the caller could easily call another image function on it.

```
detect_markers(my_mat);

detect_birds(my_mat);
// No reasonable person would expect that detect_birds will now do nonsense
```


OpenCV is a special case because `cv::Mat` is a shared_ptr, and doesn't actually let you truly const anything. So you have to "on your honor" avoid editing its contents